cypress/integration/api/course.test.ts

```diff
@@ -0,0 +1,130 @@
+import { HttpMethod } from "@lib/http-method";
+import { StatusCodes } from "http-status-codes";
+import { v4 } from "uuid";
+import {
+  TEST_COURSE_UUID
+} from "../../support/constants";
+
+describe("test course api", () => {
+  beforeEach(() => {
+    cy.task("db:setup");
+  });
+
+  afterEach(() => {
+    cy.task("db:teardown");
+  });
+
+  it("should give a 200 when GETting an existing course", () => {
+    cy.request({
+      method: HttpMethod.GET,
+      url: `/api/course/${TEST_COURSE_UUID}`,
+    }).then((response) => {
+      expect(response.status).to.equal(StatusCodes.OK);
+      expect(response.body.name).to.equal("Very Hard Course");
+      expect(response.body.code).to.equal("2H03");
+      expect(response.body.term).to.equal("Winter");
+      expect(response.body.year).to.equal(2022);
+    });
+  });
+
+  
+  it("should give a 200 when GETting posts for an existing course", () => {
+    cy.request({
+      method: HttpMethod.GET,
+      url: `/api/course/${TEST_COURSE_UUID}/posts`,
+    }).then((response) => {
+      expect(response.status).to.equal(StatusCodes.OK);
+      expect(response.body.length).to.equal(4);
+      expect(response.body[0].user).to.equal(undefined);
+      expect(response.body[1].user).to.equal(undefined);
+      expect(response.body[2].user).to.equal(undefined);
+      expect(response.body[3].user).to.equal(undefined);
+    });
+  });
+
+  it("should include user with posts when signed in when GETting an existing courses posts while AuthN", () => {
+    cy.login();
+    cy.request({
+      method: HttpMethod.GET,
+      url: `/api/course/${TEST_COURSE_UUID}/posts`,
+    }).then((response) => {
+      expect(response.status).to.equal(StatusCodes.OK);
+      expect(response.body.length).to.equal(4);
+      expect(response.body[0].user.name).to.not.equal(null)
+      expect(response.body[1].user.name).to.not.equal(null)
+      expect(response.body[2].user.name).to.not.equal(null)
+      expect(response.body[3].user.name).to.not.equal(null)
```

Comment:
better yet, for these you could probably add the name of the user, since it should just be `"Test User"` or `"Other"` I think